code/game/area/areas.dm

```diff
@@ -446,27 +446,23 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 		T = get_turf(src)
 
 	if(!T)
-		return 0
+		return FALSE
 
-	//Gravity forced on the atom
-	var/datum/component/forced_gravity/FG = GetComponent(/datum/component/forced_gravity)
-	if(FG)
-		if(!FG.ignore_space && isspaceturf(T))
-			return 0
-		else
-			return FG.gravity
-
-	//Gravity forced on the turf
-	FG = T.GetComponent(/datum/component/forced_gravity)
-	if(FG)
-		if(!FG.ignore_space && isspaceturf(T))
-			return 0
-		else
-			return FG.gravity
+	var/self_comp_flags = SEND_SIGNAL(src, COMSIG_ATOM_HAS_GRAVITY, T)
+	if(self_comp_flags & COMPONENT_FORCE_GRAVITY_ON)
+		return TRUE
+	if(self_comp_flags & COMPONENT_FORCE_GRAVITY_OFF)
+		return FALSE
+	
+	var/turf_comp_flags = SEND_SIGNAL(src, COMSIG_TURF_HAS_GRAVITY, src)
+	if(turf_comp_flags & COMPONENT_FORCE_GRAVITY_ON)
+		return TRUE
+	if(turf_comp_flags & COMPONENT_FORCE_GRAVITY_OFF)
+		return FALSE
 
 	var/area/A = get_area(T)
 	if(isspaceturf(T)) // Turf never has gravity
```

Comment:
move this if check before the `get_area()` above for a tiny perf gain in this path